### PR TITLE
Default task descriptions

### PR DESCRIPTION
**Pull request by ejensen**
This PR adds task descriptions based on the operation type, name and identifier. [Task descriptions](https://developer.apple.com/documentation/foundation/urlsessiontask/1409798-taskdescription) are helpful in disambiguation `URLSessionTask`s in Instruments and debugging in Xcode.

### CODE REVIEW

File: apollo-ios/Sources/Apollo/NetworkFetchInterceptor.swift [hunk 1]
```diff
@@ -36,7 +36,11 @@ public class NetworkFetchInterceptor: ApolloInterceptor, Cancellable {
       return
     }
     
-    let task = self.client.sendRequest(urlRequest) { [weak self] result in
+    var taskDescription = "\(Operation.operationType) \(Operation.operationName)"
+    if let operationIdentifier = Operation.operationIdentifier {
```
Comment (AnthonyMDev) [inline apollo-ios/Sources/Apollo/NetworkFetchInterceptor.swift:40]: These operation identifiers are just the id's used for persisted queries. I don't think they are really useful in this context.

**Review by AnthonyMDev — CHANGES_REQUESTED** [review]
Thanks so much for the PR! I think that we shouldn't add the `operationIdentifier` to the `taskDescription`, but otherwise, I'm good with this. If you make that change, I'm happy to merge this.

File: apollo-ios/Sources/Apollo/URLSessionClient.swift [hunk 2]
```diff
@@ -152,7 +152,7 @@ open class URLSessionClient: NSObject, URLSessionDelegate, URLSessionTaskDelegat
     sendRequest(
       request,
       taskDescription: nil,
-      rawTaskCompletionHandler: nil,
+      rawTaskCompletionHandler: rawTaskCompletionHandler,
```
[the remark this answers is not in the document]
Comment (ejensen) [reply apollo-ios/Sources/Apollo/URLSessionClient.swift:155]: It is indirectly related. I realized the `rawTaskCompletionHandler` parameter was not being forwarded correctly when overriding the `sendRequest()` method to construct a default `taskDescription` when the `taskDescription` parameter is `nil`.

File: apollo-ios/Sources/Apollo/NetworkFetchInterceptor.swift (hunk 1 above)
Comment (ejensen) [reply apollo-ios/Sources/Apollo/NetworkFetchInterceptor.swift]: Sounds good to me. I removed it in commit 897be2f

**Review by AnthonyMDev — APPROVED** [review]
Thanks for this PR @ejensen!

**Pull request merged**